contracts/Governance.sol

```diff
@@ -0,0 +1,480 @@
+  address public constant VALIDATOR_CONTRACT_ADDR = 0x0000000000000000000000000000000000001000;
+  address public constant SLASH_CONTRACT_ADDR = 0x0000000000000000000000000000000000001001;
+  address public constant SYSTEM_REWARD_ADDR = 0x0000000000000000000000000000000000001002;
+  address public constant LIGHT_CLIENT_ADDR = 0x0000000000000000000000000000000000001003;
+  address public constant TOKEN_HUB_ADDR = 0x0000000000000000000000000000000000001004;
+  address public constant INCENTIVIZE_ADDR=0x0000000000000000000000000000000000001005;
+  address public constant RELAYERHUB_CONTRACT_ADDR = 0x0000000000000000000000000000000000001006;
+  address public constant GOV_HUB_ADDR = 0x0000000000000000000000000000000000001007;
+  address public constant TOKEN_MANAGER_ADDR = 0x0000000000000000000000000000000000001008;
+  address public constant CROSS_CHAIN_CONTRACT_ADDR = 0x0000000000000000000000000000000000002000;
+  address public constant STAKING_CONTRACT_ADDR = 0x0000000000000000000000000000000000002001;
+  address public constant STAKE_HUB_ADDR = 0x0000000000000000000000000000000000002002;
+  address public constant GOVERNANCE_ADDR = 0x0000000000000000000000000000000000002003;
+
+  modifier onlyNotInit() {
+    require(!alreadyInit, "the contract already init");
+    _;
+  }
+
+  modifier onlyInit() {
+    require(alreadyInit, "the contract not init yet");
+    _;
+  }
+
+  modifier onlyGov() {
+    require(msg.sender == GOV_HUB_ADDR, "the message sender must be governance contract");
+    _;
+  }
+
+  modifier onlyGovernance() {
+    require(msg.sender == GOVERNANCE_ADDR, "the message sender must be governance v2 contract");
+    _;
+  }
+
+  modifier onlyValidatorContract() {
+    require(msg.sender == VALIDATOR_CONTRACT_ADDR, "the message sender must be validatorSet contract");
+    _;
+  }
+
+  function _compareStrings(string memory a, string memory b) internal pure returns (bool) {
+    return (keccak256(abi.encodePacked((a))) == keccak256(abi.encodePacked((b))));
+  }
+
+  function _bytesToUint256(uint _offst, bytes memory _input) internal pure returns (uint256 _output) {
+    assembly {
+      _output := mload(add(_input, _offst))
+    }
+  }
+}
+
+
+// TODO: add IGovernance extend
+contract Governance is System {
+  uint256 public constant INIT_VOTING_PERIOD = 1 weeks;
+  uint256 public constant INIT_POLL_VOTING_PERIOD = 2 weeks;
+  uint256 public constant INIT_EXECUTION_DELAY = 1 days;
+  uint256 public constant INIT_EXECUTION_EXPIRATION = 3 days;
+  uint256 public constant INIT_QUORUM_VOTING_POWER = 1e7 ether;
+  uint256 public constant INIT_POLL_SUBMIT_THRESHOLD = 50 ether;
+  uint256 public constant INIT_MIN_EXECUTE_SUPPORT_RATE = 50;
+  uint256 public constant PROPOSAL_EXECUTE_SUPPORT_RATE_SCALE = 100;
+
+  // locker => share contract => LockShare
+  mapping(address => mapping(address => ShareLock)) private lockShareMap;
+
+  Vote[] public votes;
+  // for proposal
+  ParamChangeRequest[] public paramChangeRequests;
```

Comment:
what about change to `ProposalTransaction`?

Reply:
renamed